shot/src/main/scala/com/karumi/shot/ShotPlugin.scala

```diff
@@ -61,27 +61,31 @@ class ShotPlugin extends Plugin[Project] {
   }
 
   private def addTasks(project: Project): Unit = {
+    val extension =
+      project.getExtensions.getByType[ShotExtension](classOf[ShotExtension])
     project.getTasks
       .create(RemoveScreenshotsTask.name, classOf[RemoveScreenshotsTask])
     val pullScreenshots = project.getTasks
       .create(DownloadScreenshotsTask.name, classOf[DownloadScreenshotsTask])
     val executeScreenshot = project.getTasks
       .create(ExecuteScreenshotTests.name, classOf[ExecuteScreenshotTests])
-    executeScreenshot.dependsOn(RemoveScreenshotsTask.name)
-    val extension =
-      project.getExtensions.getByType[ShotExtension](classOf[ShotExtension])
+    if (extension.runInstrumentation) {
```

Comment:
Cosmetically, this `if`-check and the next 2 `val`s could've gone inside the following `if (extension.runInstrumentation)`, but generally looks good to me. Thanks!

Reply:
Agreed! I mostly wanted to get feedback on the approach, to make sure I'm going in the right direction.